addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -1,12 +1,20 @@
 from collections import defaultdict
+from datetime import timedelta
 
-from odoo import _, fields, models
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
 from odoo.tools import SQL
 
 
 class MailTrackingDurationMixin(models.AbstractModel):
     _name = 'mail.tracking.duration.mixin'
     _description = "Mixin to compute the time a record has spent in each value a many2one field can take"
+    # (review question, todo delete)
+    # Set as inherited to permit _message_post_after_hook override (as otherwise the overrides don't seem to be taken into account).
+    # Is this how it should be done?
+    _inherit = ['mail.thread']
```

Comment:
You could check whether the method exists, as in theory it's also valid to just check if "write_date" changes for this and not care about messages if they can't happen.

But in practice we don't support tracking on anything but mail threads so it might make sense anyway.

tbd

Reply:
Yes I meant in the case it's not a thread we probably just don't consider "messages" an activity for the purposes of the stage so it would be fine to just not check it.

> how to check the hook exists

Indeed it could get tricky in practice, hence only supporting mail thread seems good enough to me for now :D